sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/statsbeat/CustomDimensions.java

```diff
@@ -19,35 +21,23 @@ public class CustomDimensions {
 
     // visible for testing
     CustomDimensions() {
-        String qualifiedSdkVersion = PropertyHelper.getQualifiedSdkVersionString();
-
-        if (qualifiedSdkVersion.startsWith("aw")) {
-            resourceProvider = ResourceProvider.RP_APPSVC;
-            operatingSystem = OperatingSystem.OS_WINDOWS;
-        } else if (qualifiedSdkVersion.startsWith("al")) {
+        if ("java".equals(System.getenv("FUNCTIONS_WORKER_RUNTIME"))) {
```

Comment:
Perhaps this code and the code below could be moved to a static method in the `ResourceProvider` enum, returning a `ResourceProvider`.